pkg/deployment/all_in_one.go

```diff
@@ -120,7 +120,7 @@ func (a *AllInOne) Get() *appsv1.Deployment {
 		},
 		Spec: appsv1.DeploymentSpec{
 			Selector: &metav1.LabelSelector{
-				MatchLabels: labels,
```

Comment:
Is `labels` being used elsewhere? I have the feeling that it should be the other way around: both should use `labels`, and `labels` should take `commonSpec.Labels` into consideration. This is because `labels` might have different values, more specific values than `commonSpec.Labels`.

Reply:
`labels` looks like it's the `default` set of recommended labels (in [utils.go](https://github.com/jaegertracing/jaeger-operator/blob/55f0191947a60e2d995353c1a6aa21ea619d39bb/pkg/util/util.go#L163)) is only used in two places:
- to initialise the `baseCommonSpec` ([L54](https://github.com/jaegertracing/jaeger-operator/blob/11a063f97c97560f1a811d95478828b68c7aeba1/pkg/deployment/all_in_one.go#L54)) which appears to be the `default` spec.
- and this line in question, which is to use this `default` set of labels as the label selector.

The more specific set of labels is actually `commonSpec.Labels` as it is the result of merging the `labels` with the custom set of labels defined in `a.jaeger.Spec.AllInOne.JaegerCommonSpec` and `a.jaeger.Spec.JaegerCommonSpec` (see [L57](https://github.com/jaegertracing/jaeger-operator/blob/11a063f97c97560f1a811d95478828b68c7aeba1/pkg/deployment/all_in_one.go#L57)).

If you agree with the above, it does raise an interesting point that perhaps the variable naming could be improved as I would agree the name `labels` sounds more specific than `commonSpec.Labels`. Perhaps `labels` could be renamed to `defaultLabels` or `recommendedLabels` and `commonSpec` -> `mergedSpec`?

Reply:
If this line is being changed to use `commonSpec.Labels`, then we don't need `labels` as a var anymore. We could use `a.labels()` directly in [L54](https://github.com/jaegertracing/jaeger-operator/blob/11a063f97c97560f1a811d95478828b68c7aeba1/pkg/deployment/all_in_one.go#L54)